clippy_utils/src/numeric_literal.rs

```diff
@@ -219,7 +219,7 @@ impl<'a> NumericLiteral<'a> {
 }
 
 fn split_suffix<'a>(src: &'a str, lit_kind: &LitKind) -> (&'a str, Option<&'a str>) {
-    debug_assert!(lit_kind.is_numeric());
+    debug_assert!(lit_kind.is_numeric(), "`lit_kind` should be numeric");
     lit_suffix_length(lit_kind)
```

Comment:
My two cents: these changes display the very problem with this lint.
It encourages people to just add *some kind* of assert-message to silent the lint, but the message provides zero information.

If `foo.is_numeric()` returns `false` and blows up an assert, I already know that `foo` is not numeric and that this is problematic.
Why would would I want to waste time and add the same exact information "foo should be numeric" again, in a textual form (which just adds additional maintenance burden because now I have to additionally update the message every time I change the variable name, etc..) ?

IMO it would be good to add an additional message to the lint, something along the lines of
`help: try to describe why the failing assert is problematic`. The inserted assert-message should be more about the `why is it important to know that this failed` and not `what exactly failed here`, because the code inside the assert already conveys that.

Reply:
I agree that the messages provided here are a bit meh. However, I would not say that this is a problem with the lint; we simply have no way of knowing whether an assert message will satisfy. However, having them at all will at least put them under the eyes of a potential reviewer. Also I think your point should be added to the lint's documentation.

Perhaps tne lint should be put in `restriction` instead of `pedantic`.

Reply:
Yes you're right about the messages, since we are not very familiar with the codebase we tried to come up with messages by quickly looking at the assertion but it was hard to come up with good messages. 
I've added a help message and updated the lint's description. 
Didn't know about `restriction`, it makes more sense for this lint. I've reverted the "dogfood" commit, I believe someone with more familiarity with the codebase can enable the lint and come up with good assert messages in a follow-up PR.